Automattic/woocommerce-payments
# Disable the manual refunds button unless a WCPay refund has failed

Supercedes #710

Fixes #426

#### Changes proposed in this Pull Request

- Save the refund status in `_wcpay_refund_status` as a meta value.
- Use the meta value to conditionally disable the manual refund button.

Context: Payments made through WooCommerce Payments must be refunded through WCPay. Only after the refund fails may the funds be refunded manually.

#### Testing instructions

1. `git checkout 426-display-manual-refund-after-failure`
2. `npx webpack`
1. Make a purchase with test card `4000 0000 0000 5126`.
2. Navigate to the order screen and click "Refund" to display the refund bar.
3. Make sure that the button for manual refunds is disabled and explains that "Refunds are available only through WooCommerce Payments."
4. Attempt a refund through WCPay.
5. The refund should go through "successfully".
6. Reload the page.
7. Note that there is now a new note explaining that the refund actually failed.
8. Delete the refund by clicking the `x` button that appears when you hover over the refund entry.
9. Try making another refund.
10. Note that there is now only a manual refund option available.
